src/topp/QualityControl.cpp

```diff
@@ -420,7 +430,47 @@ class TOPPQualityControl : public TOPPBase
 
       if (export_evidence.isValid())
       {
-        export_evidence.exportFeatureMap(*fmap,cmap);
+        auto& cmapProtIds = cmap.getProteinIdentifications();
+        map<String,String> fastaMap {};
+        //map the identifier to the description so that we can access the description via the cmap-identifier
+        for(const auto& entry : protDescription)
+        {
+          fastaMap.emplace(entry.identifier, entry.description);
+        }
+
+        if(cmapProtIds.empty())
+        {
+          OPENMS_LOG_ERROR << "No ProteinIdentifications in consensusmap... aboarting" << std::endl;
+        }
+
+        // THIS IS OUR FALLBACK IF THE USER PROVIDED NO .fasta
+        if(protDescription.empty())
+        {
+          OPENMS_LOG_INFO << "No FASTA passed, looking for the default search parameters in consensusXML" << std::endl;
+          const String& fileName = cmapProtIds[0].getSearchParameters().db;
+          FASTAFile().load(fileName, protDescription);
+        }
```

Comment:
too late here.
If you load the FASTA file now, your index will be empty... move this code up to line 184